src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPECIFIED_STRING_S = "supply";
+    private static final String SPECIFIED_STRING_B = "buy";
+    private static final String SPECIFIED_STRING_R = "result";
+    private static final String SEPARATOR = ",";
+    private static final int VALUE_ONE = 0;
+    private static final int VALUE_TWO = 1;
+    private static final int VALUE_THREE = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFile(fromFileName);
+        String report = generateReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        StringBuilder data = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                data.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Better to have a good error description
```suggestion
            throw new RuntimeException("Unable to read file: "+ fromFileName, e);
```